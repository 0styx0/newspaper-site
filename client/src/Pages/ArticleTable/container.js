import React from 'react';
import fetchFromApi from '../../helpers/fetchFromApi';

import ArticleTable from './';

export default class ArticleTableContainer extends React.Component {

    constructor() {
        super();

        this.putData = this.putData.bind(this);

        this.state = {
            history: []
        };
    }

    async putData(num = null) {

        const lastPath = +window.location.pathname.split("/").slice(-1);

        num = (isNaN(lastPath) || ((!isNaN(num)) && num)) ? num : lastPath;


        // use cached results (don't load new info)
        if (this.state.history[num]) {
            this.setState({
                articles: this.state.history[num].articles,
                issueInfo: this.state.history[num].issueInfo
            });

            window.history.pushState({}, `Issue ${num}`, num);
        }
        else {

            const data = await this.getData(num);
            const json = await data.json();

            window.history.pushState({}, `Issue ${num}`, isNaN(lastPath) ? "modifyArticles/"+json[2].num : json[2].num);

            this.setArticleInfo(json);
        }
    }

    componentWillMount() {

        this.putData(null);
    }

    async getData(num) {

         return await fetchFromApi(`articleGroup?articlesFor=${num}`);

    }


    setArticleInfo(data) {

        const history = [...this.state.history];
        history[data[2].num] = {articles: data[0], issueInfo: data[2]};

        this.setState({
            articles: data[0].length ? data[0] :  null,
            issueInfo: data[2],
            history
        });
    }

    render() {

            return <ArticleTable
                     articles={this.state.articles}
                     issue={this.state.issueInfo}
                     update={(e) => this.putData(e.target.value)}
                   />
    }
}
