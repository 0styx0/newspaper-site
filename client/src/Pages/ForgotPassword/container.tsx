import * as React from 'react';
import { graphql, withApollo } from 'react-apollo';
import { RecoverPassword } from '../../graphql/user';
import ForgotPassword from './';

interface Props {
    recoverPassword:
      (params: {
          query: typeof RecoverPassword,
          variables: { username: string, authCode: string, email: string }
        }
      ) => Promise<{message: string}>;
}

export class ForgotPasswordContainer extends React.Component<Props, {}> {

    constructor(props: Props) {
        super(props);

        this.onSubmit = this.onSubmit.bind(this);

    }

    /**
     * Sends data to server
     */
    onSubmit(target: HTMLFormElement) {

        this.props.recoverPassword({
            query: RecoverPassword,
            variables: {
                authCode: (target.querySelector('input[name=lastAuth]') as HTMLInputElement).value,
                username: (target.querySelector('input[name=username]') as HTMLInputElement).value,
                email: (target.querySelector('input[name=email]') as HTMLInputElement).value,
            }
        });
    }

    render() {
        return <ForgotPassword onSubmit={this.onSubmit} />;
    }
}

const ForgotPasswordContainerWithData =
    graphql(RecoverPassword, {name: 'recoverPassword'})(ForgotPasswordContainer as any);

export default withApollo(ForgotPasswordContainerWithData);
