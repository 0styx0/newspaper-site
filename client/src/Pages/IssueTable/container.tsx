import * as React from 'react';
import { compose, graphql } from 'react-apollo';
import { IssueQuery, IssueUpdate } from '../../graphql/issues';
import { Issue } from './interface.shared';
import IssueTable from './';

interface State {
    privateIssue: { // admins can change these (until public is true)
        public?: boolean;
        name?: string;
    };
    loaded: boolean;
}

interface Props {
    data: {
        loading: boolean;
        issues: Issue[]
    };
    mutate: Function;
}

/**
 * Creates a table of issues, in descending order with info of
 * number, name, datePublished, and views (sum of views of all articles in the issue)
 *
 * If user is admin (lvl 3) and issue is not public, user can change issue name and make it public.
 * In that case, @see IssueTable.allowEditsOfLastIssue for difference in UI
 */
export class IssueTableContainer extends React.Component<Props, State> {

    constructor() {
        super();

        this.onChangeIssueInfo = this.onChangeIssueInfo.bind(this);
        this.onSubmit = this.onSubmit.bind(this);

        this.state = {
            privateIssue: {},
            loaded: false
        };
    }

    /**
     * When graphql gets data from @see graphql/issues,
     * this turns it into 2d array of what will be table rows and saves it to state.issueInfo
     *
     * If current user is an admin, enables editing of most recent, unpublished issue @see this.allowEditsOfLastIssue
     */
    componentWillReceiveProps(props: Props) {

        if (!props.data.issues || this.props.data.issues.length > 0) {
            return;
        }

        this.setState({loaded: true});
    }

    /**
     * @param e {HTMLInputElement event}
     *
     * Adds input's value to this.state.privateIssue[input.name]
     *
     * @example if event.target = <input name="public" value="1" />, then after this, this.state.privateIssue.public = 1
     */
    onChangeIssueInfo(e: Event) {

        const target = e.target as HTMLInputElement;

        const privateIssue = Object.assign({}, this.state.privateIssue);
        privateIssue[target.name] = target.value;

        this.setState({
            privateIssue
        });
    }

    /**
     * @param e {HTMLFormElement event}
     *
     * Sends this.state.privateIssue data to server (so name and/or public status can be saved to db)
     */
    onSubmit(e: Event) {

        e.stopPropagation();
        e.preventDefault();

        this.props.mutate({
            variables: {
                name: this.state.privateIssue.name,
                public: !!this.state.privateIssue.public
            }
        });
    }

    render() {

        if (this.props.data.issues.length < 1) {
            return null;
        }

        return (
            <IssueTable
              key={+this.state.loaded}
              onSubmit={this.onSubmit}
              onChangeIssueInfo={this.onChangeIssueInfo}
              issues={this.props.data.issues}
            />
        );
    }
}


const IssueTableContainerWithData = compose(
    graphql(IssueQuery),
    graphql(IssueUpdate)
)(IssueTable as any);

export default IssueTableContainerWithData;