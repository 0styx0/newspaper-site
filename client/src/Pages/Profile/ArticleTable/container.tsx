import * as React from 'react';
import { graphql, withApollo, compose } from 'react-apollo';
import { ArticleDelete } from '../../../graphql/articles';
import { Article } from '../shared.interfaces';
import UserArticleTable from './';

interface Props {
    articles: Article[];
    deleteArticle: Function;
}

/**
 * Container for @see ./index.tsx
 */
export class UserArticleTableContainer extends React.Component<Props, {idsToDelete: Set<string>}> {

    constructor() {
        super();

        this.onDelete = this.onDelete.bind(this);
        this.onSubmit = this.onSubmit.bind(this);

        this.state = {
            idsToDelete: new Set<string>()
        };
    }

    /**
     * Adds an article to state.idsToDelete
     */
    onDelete(e: Event) {

        const idsToDelete: Set<string> = this.state.idsToDelete;
        const value = (e.target as HTMLInputElement).value;

        if (idsToDelete.has(value)) {

            idsToDelete.delete(value);
        } else {

            idsToDelete.add(value);
        }

        this.setState({
            idsToDelete
        });
    }

    /**
     * Sends idsToDelete to server to be deleted
     */
    onSubmit(e: Event) {

        e.preventDefault();
        e.stopPropagation();

        this.props.deleteArticle({
            variables: {
                ids: [...this.state.idsToDelete]
            }
        });
    }

    render() {

        return (
            <UserArticleTable
              articles={this.props.articles}
              onDelete={this.onDelete}
              onSubmit={this.onSubmit}
            />
        );
    }
}


const UserArticleTableContainerWithData = compose(
    graphql(ArticleDelete, {name: 'deleteArticle'})
)(UserArticleTableContainer as any);

export default withApollo(UserArticleTableContainerWithData);