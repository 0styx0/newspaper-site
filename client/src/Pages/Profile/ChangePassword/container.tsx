import * as React from 'react';
import ChangePassword from './';
import { UserUpdate } from '../../../graphql/user';
import { graphql, compose, withApollo } from 'react-apollo';

interface Props {
    updatePassword: Function;
}

export class ChangePasswordContainer extends React.Component<Props, {}> {

    constructor() {
        super();

        this.onSubmit = this.onSubmit.bind(this);
    }

    /**
     * Send data to server
     */
    onSubmit(e: Event) {

        const target = e.target as HTMLFormElement;

        const oldPassword = target.querySelector('[name=password]') as HTMLInputElement;
        const newPassword = target.querySelector('[name=newPassword]') as HTMLInputElement;
        const newPasswordConfirmation = target.querySelector('[name=newPasswordConfirmation]') as HTMLInputElement;

        if (newPassword.value !== newPasswordConfirmation.value ||
          !(oldPassword.value && newPassword.value && newPasswordConfirmation.value)) {

            return;
        }

        this.props.updatePassword({
            variables: {
                password: oldPassword.value,
                newPassword: newPassword.value
            }
        });
    }

    render() {
        return <ChangePassword onSubmit={this.onSubmit} />;
    }
}

const ChangePasswordContainerWithData = compose(
    graphql(UserUpdate, {name: 'updatePassword'}) as any
)(ChangePasswordContainer);

export default withApollo(ChangePasswordContainerWithData);