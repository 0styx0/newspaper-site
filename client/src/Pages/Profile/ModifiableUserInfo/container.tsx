import * as React from 'react';
import { PrivateUserQuery, UserUpdate } from '../../../graphql/user';
import { UserDelete } from '../../../graphql/users';
import { graphql, withApollo, compose } from 'react-apollo';
import { ModifiableUserInfo } from '../shared.interfaces';
import ModifiableUserInfoComponent from './';

interface Props {
    updateUser: Function;
    deleteUser: Function;
    privateUserData: {
        users: [ModifiableUserInfo] // always length = 1
    };
}

interface State {
    updates: {
        twoFactor?: boolean;
        notificationStatus?: boolean;
    };
    delete?: boolean;
}

/**
 * Container, has event listeners and passes data to @see ./index.tsx
 */
export class ModifiableUserInfoContainer extends React.Component<Props, State> {

    constructor() {
        super();

        this.onDelete = this.onDelete.bind(this);
        this.onSubmit = this.onSubmit.bind(this);
        this.onChange = this.onChange.bind(this);

        this.state = {
            updates: {}
        };
    }

    onDelete(e: Event) {

        this.setState({
            delete: (e.target as HTMLInputElement).checked
        });
    }

    /**
     * Saves changes to state.updates
     *
     * @uses `e.target.name`, `e.target.value`
     */
    onChange(e: Event) {

        const target = e.target as HTMLInputElement;

        const updates = Object.assign({}, this.state.updates);

        updates[target.name] = target.checked;

        this.setState({
            updates
        });
    }

    /**
     * Sends `state.updates` to server
     */
    onSubmit(target: HTMLFormElement) {

        if (this.state.delete) {
            return this.deleteUser();
        }

        if (Object.keys(this.state.updates).length > 0) {
            this.props.updateUser({
                variables: Object.assign(this.state.updates, {
                    password: (target.querySelector('[name=password]') as HTMLInputElement).value
                })
            });
        }
    }

    /**
     * Sends id of user to server so user can be deleted
     */
    deleteUser() {

        this.props.deleteUser({
            variables: {
                ids: [this.props.privateUserData.users[0].id]
            }
        });
    }

    render() {

        if (!this.props.privateUserData || !this.props.privateUserData.users) {
           return null;
        }

        return (
            <ModifiableUserInfoComponent
              onSubmit={this.onSubmit}
              onChange={this.onChange}
              onDelete={this.onDelete}
              {...this.props.privateUserData.users[0]}
            />
        );
    }
}

const ModifiableUserInfoContainerWithData = compose(
    graphql(PrivateUserQuery, {
        name: 'privateUserData',
        options: {
            variables: {
                profileLink: 'meiselesd2018'
            }
        }
    }),
    graphql(UserUpdate, {name: 'updateUser'}),
    graphql(UserDelete, {name: 'deleteUser'})
)(ModifiableUserInfoContainer as any);

export default withApollo(ModifiableUserInfoContainerWithData);
