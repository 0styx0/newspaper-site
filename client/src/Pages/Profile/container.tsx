import * as React from 'react';
import { UserQuery } from '../../graphql/user';
import { graphql, compose, withApollo } from 'react-apollo';
import Profile from './';
import { Article, PublicUserInfo } from './shared.interfaces';

interface Props {
    data: {
        users: {
            articles: Article[];
        } & PublicUserInfo
    };
    client: {
        query: ( params: { query: typeof UserQuery, variables: { profileLink: string; } } ) => Promise<Props>;
    };
}

interface State {
    articles: Article[];
    user: PublicUserInfo;
}

class ProfileContainer extends React.Component<Props, State> {

    public state: State;

    constructor(props: Props) {
        super(props);

        this.state = {
            user: {} as PublicUserInfo,
            articles: [] as Article[]
        };

    }

    async componentWillMount() {

        const path = window.location.pathname.split('/');

        const { data } = await this.props.client.query({
            query: UserQuery,
            variables: {
                profileLink: path[path.length - 1]
            }
        });

        this.setState({
            user: data.users[0],
            articles: data.users[0].articles
        });
    }

    render() {

        if (!this.state.user) {
            return null;
        }

        const articles = this.state.articles as Article[];
        const user = this.state.user as PublicUserInfo;

        return (
            <Profile
                articles={articles}
                user={user}
            />
        );
    }
}

const ProfileContainerWithData = compose(
    graphql(UserQuery, {
        options: {
            variables: {
                profileLink: window.location.pathname.split('/')[window.location.pathname.split('/').length - 1]
            }
        }
    })
)(ProfileContainer);

export default withApollo(ProfileContainerWithData as any) as any;
