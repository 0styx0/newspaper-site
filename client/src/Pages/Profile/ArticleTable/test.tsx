import * as React from 'react';
import { UserArticleTableContainer } from './container';
import { mount } from 'enzyme';
import * as renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import localStorageMock from '../../../tests/localstorage.mock';
import casual from '../casual.data';
import snapData from './articles.example';
import { randomCheckboxToggle } from '../../../tests/enzyme.helpers';
import toggler from '../../../helpers/toggler';

localStorageMock.setItem('jwt', JSON.stringify([, {level: 3}]));

const data = {
    articles: casual.articles(casual.integer(0, 25)) // if do more than around 25, the test will be really slow
};

function setup(mockGraphql: {deleteArticle?: Function} = {}) {

    return mount(
        <MemoryRouter>
            <UserArticleTableContainer
                articles={data.articles}
                deleteArticle={mockGraphql.deleteArticle ? mockGraphql.deleteArticle : (test: {}) => false}
                canModify={true}
            />
        </MemoryRouter>
    );
}

fdescribe('<UserArticleTableContainer>', () => {

    describe('snapshots', () => {

        function testSnap(canModify: boolean) {

            const tree = renderer.create(

                <MemoryRouter>
                    <UserArticleTableContainer
                        articles={snapData}
                        deleteArticle={(test: {}) => false}
                        canModify={canModify}
                    />
                </MemoryRouter>
            ).toJSON();

            expect(tree).toMatchSnapshot();

        }

        it('renders correctly when canModify is true', () => testSnap(true));

        it('renders correctly when canModify is false', () => testSnap(false));
    });

    describe('onDelete', () => {

        let wrapper: any;
        let component: any;
        let deleteBoxes: any;

        beforeEach(() => {

            wrapper = setup();

            deleteBoxes = wrapper.find('[name="delArt"]');
            component = wrapper.find(UserArticleTableContainer).node;
        });

        it('adds article id to state.idsToDelete when checkbox is clicked', () => {

            let expectedIds = new Set<string>();
            let articlesToTest = casual.integer(0, deleteBoxes.length - 1);

            for (let i = 0; expectedIds.size < articlesToTest; i++) {

                const result = randomCheckboxToggle(deleteBoxes);

                toggler(expectedIds, component.props.articles[result.index].id);
            }

            expect([...component.state.idsToDelete]).toEqual([...expectedIds]);
        });

        it('removes article id from state.idsToDelete when checkbox is unchecked', () => {

            let expectedIds = new Set<string>();
            let indices = new Set<String>();
            let articlesToTest = casual.integer(0, deleteBoxes.length - 1);

            for (let i = 0; expectedIds.size < articlesToTest; i++) {

                const result = randomCheckboxToggle(deleteBoxes);
                const id = result.input.nodes[0].value;

                toggler(expectedIds, id);
                toggler(indices, result.index);
            }

            for (let i = 0; i < casual.integer(0, indices.size); i++) {

                const indexToRemove = casual.random_element([...indices]);

                const result = randomCheckboxToggle(deleteBoxes, indexToRemove);
                indices.delete(indexToRemove);

                expectedIds.delete(result.input.nodes[0].value);
            }

            expect([...component.state.idsToDelete].sort()).toEqual([...expectedIds].sort());
        });
    });
});
