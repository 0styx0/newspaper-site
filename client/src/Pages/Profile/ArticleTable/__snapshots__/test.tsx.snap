// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<UserArticleTableContainer> snapshots renders correctly when canModify is false 1`] = `
<section
  className="container "
>
  <h1>
    Articles
  </h1>
  <form
    onSubmit={[Function]}
  >
    <table>
      <thead>
        <tr>
          <th>
             
            Article
          </th>
          <th>
             
            Published
          </th>
          <th>
             
            Type
          </th>
          <th>
             
            Views
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <a
              href="/issue/90/story/A%20nam"
              onClick={[Function]}
            >
              A nam
            </a>
          </td>
          <td>
            1990-05-14
          </td>
          <td>
            quia
          </td>
          <td>
            30
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/27/story/Sint%20et"
              onClick={[Function]}
            >
              Sint et
            </a>
          </td>
          <td>
            1985-07-13
          </td>
          <td>
            nobis, et
          </td>
          <td>
            17
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/56/story/Dolor%20qui"
              onClick={[Function]}
            >
              Dolor qui
            </a>
          </td>
          <td>
            1974-11-22
          </td>
          <td>
            velit, tenetur, voluptate
          </td>
          <td>
            83
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/99/story/Ut%20facilis"
              onClick={[Function]}
            >
              Ut facilis
            </a>
          </td>
          <td>
            1972-02-25
          </td>
          <td>
            facere, quia
          </td>
          <td>
            2
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/21/story/Animi%20iure"
              onClick={[Function]}
            >
              Animi iure
            </a>
          </td>
          <td>
            2002-07-07
          </td>
          <td>
            sint
          </td>
          <td>
            70
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/2/story/Nihil%20nisi"
              onClick={[Function]}
            >
              Nihil nisi
            </a>
          </td>
          <td>
            2012-09-11
          </td>
          <td>
            sed, necessitatibus
          </td>
          <td>
            24
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/78/story/A%20consequatur%20doloremque"
              onClick={[Function]}
            >
              A consequatur doloremque
            </a>
          </td>
          <td>
            1989-04-13
          </td>
          <td>
            quo, consequatur
          </td>
          <td>
            6
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/14/story/Illo%20est%20sed"
              onClick={[Function]}
            >
              Illo est sed
            </a>
          </td>
          <td>
            1980-08-18
          </td>
          <td>
            dolorum, sint, rerum
          </td>
          <td>
            84
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/88/story/Repudiandae%20asperiores%20repellat"
              onClick={[Function]}
            >
              Repudiandae asperiores repellat
            </a>
          </td>
          <td>
            1994-07-27
          </td>
          <td>
            quis, itaque, harum
          </td>
          <td>
            98
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/14/story/Neque%20ad"
              onClick={[Function]}
            >
              Neque ad
            </a>
          </td>
          <td>
            1987-06-14
          </td>
          <td>
            in, consequatur
          </td>
          <td>
            13
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/25/story/Dicta%20at%20harum"
              onClick={[Function]}
            >
              Dicta at harum
            </a>
          </td>
          <td>
            1985-10-01
          </td>
          <td>
            velit, recusandae
          </td>
          <td>
            78
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/69/story/Quo%20at"
              onClick={[Function]}
            >
              Quo at
            </a>
          </td>
          <td>
            2010-04-08
          </td>
          <td>
            et, laudantium
          </td>
          <td>
            29
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/11/story/Ut%20ipsam"
              onClick={[Function]}
            >
              Ut ipsam
            </a>
          </td>
          <td>
            2011-05-03
          </td>
          <td>
            mollitia
          </td>
          <td>
            3
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/85/story/Consequuntur%20placeat"
              onClick={[Function]}
            >
              Consequuntur placeat
            </a>
          </td>
          <td>
            1990-01-27
          </td>
          <td>
            expedita, temporibus
          </td>
          <td>
            17
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/92/story/Possimus%20dolorem%20molestias"
              onClick={[Function]}
            >
              Possimus dolorem molestias
            </a>
          </td>
          <td>
            1996-04-12
          </td>
          <td>
            dignissimos, molestias
          </td>
          <td>
            19
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/18/story/Illum%20quo%20consequatur"
              onClick={[Function]}
            >
              Illum quo consequatur
            </a>
          </td>
          <td>
            1982-05-03
          </td>
          <td>
            qui, a
          </td>
          <td>
            58
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/82/story/Hic%20laborum"
              onClick={[Function]}
            >
              Hic laborum
            </a>
          </td>
          <td>
            2014-06-30
          </td>
          <td>
            non, iure
          </td>
          <td>
            72
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/8/story/Laborum%20omnis%20quam"
              onClick={[Function]}
            >
              Laborum omnis quam
            </a>
          </td>
          <td>
            2009-04-21
          </td>
          <td>
            enim, et
          </td>
          <td>
            78
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/4/story/Voluptatem%20accusantium"
              onClick={[Function]}
            >
              Voluptatem accusantium
            </a>
          </td>
          <td>
            1973-07-28
          </td>
          <td>
            saepe
          </td>
          <td>
            96
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/40/story/Exercitationem%20velit"
              onClick={[Function]}
            >
              Exercitationem velit
            </a>
          </td>
          <td>
            1974-01-09
          </td>
          <td>
            vel, quis
          </td>
          <td>
            69
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/20/story/Libero%20voluptas"
              onClick={[Function]}
            >
              Libero voluptas
            </a>
          </td>
          <td>
            1971-07-26
          </td>
          <td>
            odit, eum
          </td>
          <td>
            1
          </td>
        </tr>
      </tbody>
    </table>
    
  </form>
</section>
`;

exports[`<UserArticleTableContainer> snapshots renders correctly when canModify is true 1`] = `
<section
  className="container "
>
  <h1>
    Articles
  </h1>
  <form
    onSubmit={[Function]}
  >
    <table>
      <thead>
        <tr>
          <th>
             
            Article
          </th>
          <th>
             
            Published
          </th>
          <th>
             
            Type
          </th>
          <th>
             
            Views
          </th>
          <th>
             
            <span
              className="danger"
            >
              Delete
            </span>
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            <a
              href="/issue/90/story/A%20nam"
              onClick={[Function]}
            >
              A nam
            </a>
          </td>
          <td>
            1990-05-14
          </td>
          <td>
            quia
          </td>
          <td>
            30
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="corporis--20"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/27/story/Sint%20et"
              onClick={[Function]}
            >
              Sint et
            </a>
          </td>
          <td>
            1985-07-13
          </td>
          <td>
            nobis, et
          </td>
          <td>
            17
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="ipsa--19"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/56/story/Dolor%20qui"
              onClick={[Function]}
            >
              Dolor qui
            </a>
          </td>
          <td>
            1974-11-22
          </td>
          <td>
            velit, tenetur, voluptate
          </td>
          <td>
            83
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="doloribus--18"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/99/story/Ut%20facilis"
              onClick={[Function]}
            >
              Ut facilis
            </a>
          </td>
          <td>
            1972-02-25
          </td>
          <td>
            facere, quia
          </td>
          <td>
            2
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="ut--17"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/21/story/Animi%20iure"
              onClick={[Function]}
            >
              Animi iure
            </a>
          </td>
          <td>
            2002-07-07
          </td>
          <td>
            sint
          </td>
          <td>
            70
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="aliquid--16"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/2/story/Nihil%20nisi"
              onClick={[Function]}
            >
              Nihil nisi
            </a>
          </td>
          <td>
            2012-09-11
          </td>
          <td>
            sed, necessitatibus
          </td>
          <td>
            24
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="maxime--15"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/78/story/A%20consequatur%20doloremque"
              onClick={[Function]}
            >
              A consequatur doloremque
            </a>
          </td>
          <td>
            1989-04-13
          </td>
          <td>
            quo, consequatur
          </td>
          <td>
            6
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="nesciunt--14"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/14/story/Illo%20est%20sed"
              onClick={[Function]}
            >
              Illo est sed
            </a>
          </td>
          <td>
            1980-08-18
          </td>
          <td>
            dolorum, sint, rerum
          </td>
          <td>
            84
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="fuga--13"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/88/story/Repudiandae%20asperiores%20repellat"
              onClick={[Function]}
            >
              Repudiandae asperiores repellat
            </a>
          </td>
          <td>
            1994-07-27
          </td>
          <td>
            quis, itaque, harum
          </td>
          <td>
            98
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="non--12"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/14/story/Neque%20ad"
              onClick={[Function]}
            >
              Neque ad
            </a>
          </td>
          <td>
            1987-06-14
          </td>
          <td>
            in, consequatur
          </td>
          <td>
            13
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="inventore--11"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/25/story/Dicta%20at%20harum"
              onClick={[Function]}
            >
              Dicta at harum
            </a>
          </td>
          <td>
            1985-10-01
          </td>
          <td>
            velit, recusandae
          </td>
          <td>
            78
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="alias--10"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/69/story/Quo%20at"
              onClick={[Function]}
            >
              Quo at
            </a>
          </td>
          <td>
            2010-04-08
          </td>
          <td>
            et, laudantium
          </td>
          <td>
            29
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="ut--9"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/11/story/Ut%20ipsam"
              onClick={[Function]}
            >
              Ut ipsam
            </a>
          </td>
          <td>
            2011-05-03
          </td>
          <td>
            mollitia
          </td>
          <td>
            3
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="dolores--8"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/85/story/Consequuntur%20placeat"
              onClick={[Function]}
            >
              Consequuntur placeat
            </a>
          </td>
          <td>
            1990-01-27
          </td>
          <td>
            expedita, temporibus
          </td>
          <td>
            17
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="qui--7"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/92/story/Possimus%20dolorem%20molestias"
              onClick={[Function]}
            >
              Possimus dolorem molestias
            </a>
          </td>
          <td>
            1996-04-12
          </td>
          <td>
            dignissimos, molestias
          </td>
          <td>
            19
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="inventore--6"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/18/story/Illum%20quo%20consequatur"
              onClick={[Function]}
            >
              Illum quo consequatur
            </a>
          </td>
          <td>
            1982-05-03
          </td>
          <td>
            qui, a
          </td>
          <td>
            58
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="repudiandae--5"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/82/story/Hic%20laborum"
              onClick={[Function]}
            >
              Hic laborum
            </a>
          </td>
          <td>
            2014-06-30
          </td>
          <td>
            non, iure
          </td>
          <td>
            72
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="minima--4"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/8/story/Laborum%20omnis%20quam"
              onClick={[Function]}
            >
              Laborum omnis quam
            </a>
          </td>
          <td>
            2009-04-21
          </td>
          <td>
            enim, et
          </td>
          <td>
            78
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="molestiae--3"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/4/story/Voluptatem%20accusantium"
              onClick={[Function]}
            >
              Voluptatem accusantium
            </a>
          </td>
          <td>
            1973-07-28
          </td>
          <td>
            saepe
          </td>
          <td>
            96
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="labore--2"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/40/story/Exercitationem%20velit"
              onClick={[Function]}
            >
              Exercitationem velit
            </a>
          </td>
          <td>
            1974-01-09
          </td>
          <td>
            vel, quis
          </td>
          <td>
            69
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="non--1"
            />
          </td>
        </tr>
        <tr>
          <td>
            <a
              href="/issue/20/story/Libero%20voluptas"
              onClick={[Function]}
            >
              Libero voluptas
            </a>
          </td>
          <td>
            1971-07-26
          </td>
          <td>
            odit, eum
          </td>
          <td>
            1
          </td>
          <td>
            <input
              name="delArt"
              onChange={[Function]}
              type="checkbox"
              value="eos--0"
            />
          </td>
        </tr>
      </tbody>
    </table>
    <div>
      <label>
        Password: 
        <input
          name="password"
          required={true}
          type="password"
        />
        <span
          className="danger"
        >
          *
        </span>
        
      </label>
      <input
        type="submit"
      />
    </div>
  </form>
</section>
`;
